### DISCUSSION TITLE
Change Defaults of Displaying Files/Dirs

### DISCUSSION THREAD

**Discussion by asim215** [sxyazi/yazi#741] [5. Polls]
Changing default way of display files/dirs to more commonly used. 

- sorting: alphabetical -> natural
- sort_dir_first: false -> true
- show_hidden: false -> true

**Option 1: after change**:
- sorting: alphabetical -> natural
- sort_dir_first: false -> true
- show_hidden: false -> true
After:
![20240227_13h41m26s_grim](https://github.com/sxyazi/yazi/assets/7127165/0092130c-80f0-4b50-bd9f-6c2063cc3179)

**Option 2: original**. :
Before, `show_hidden=false` (original):
![20240227_13h45m13s_grim](https://github.com/sxyazi/yazi/assets/7127165/ad6d7694-b6ea-4d1f-9609-e9551150df82)
I think most of the time we need to see hidden files.

**Option 3: original with show hidden**:
- show_hidden: false -> true
Before, with change `show_hidden=true`:
![20240227_13h44m55s_grim](https://github.com/sxyazi/yazi/assets/7127165/87a0e299-e5d4-4db2-83e6-f4b1c51611f0)
Files and dirs clashed together and not separated.

**Comment by asim215** [comment]
#### Summary:
- Option 1 (proposed #739):
```sorting: natural, sort_dir_first: true, show_hidden: true```
- Option 2 (original):
```sorting: alphabetical, sort_dir_first: false, show_hidden: false```
- Option 3 (original with show_hidden=true):
```sorting: alphabetical, sort_dir_first: false, show_hidden: true```
- Option 4: other

Other options proposed:
@nairby
```
sorting: alphabetical, sort_dir_first: true, show_hidden: false
```
@Sonico98 
```
sorting: natural, sort_dir_first: true, show_hidden: false
```

**Reply by Sonico98** [reply]
I would go for Option 1 but with show_hidden: false.
I believe hidden files are, well, hidden, for a reason... Specially on other systems like Windows.

**Comment by asim215** [comment]
Here difference between alphabetical and natural.

sorting alphabetical:
![20240227_17h23m50s_grim](https://github.com/sxyazi/yazi/assets/7127165/41c3b41b-a4cd-4a5a-9584-4a53772b707a)

sorting natural:
![20240227_17h26m56s_grim](https://github.com/sxyazi/yazi/assets/7127165/824e43dc-097e-43a5-b9e9-b1e029733b69)

**Comment by sxyazi** [comment]
Sorry, but I'll choose to keep the current behavior for the following reasons:

1. I want to keep Yazi's behavior as consistent as possible with `ls`.
2. Natural sorting, due to its algorithm, requires more computational resources than `alphabetical` sorting. Even though Yazi's implementation of `natural` is [~6 times faster than the `natord` that `eza` uses in case-insensitive sorting](https://github.com/sxyazi/yazi/pull/237), it is still slower than `alphabetical`. And some users may not actually notice the difference between `natural` and `alphabetical` as the results are very similar. However, setting `natural` as the default would require these users to pay an additional performance cost and if they realize they need it, I encourage configuring it manually.
3. `sort_dir_first` makes files always appear after all directories. When a directory contains a large number of subdirectories, this can make it more difficult for users to find files.
4. `show_hidden` - IIRC, most file managers hide hidden files by default. Yazi has many users who are not computer professionals and are unlikely to need to see these hidden files. For professional users, changing this setting is also easy.
